contracts/poap-manager/src/msg.rs

```diff
@@ -0,0 +1,143 @@
+use cosmwasm_std::{Addr, Uint64};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::error::ContractError;
+use poap::msg::InstantiateMsg as POAPInstantiateMsg;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {
+    /// Address of who will have the right to administer the contract.
+    pub admin: String,
+    /// Id of the POAP contract to initialize together with this contract.
+    pub poap_code_id: Uint64,
+    /// Initialization message that will be sent to the POAP contract.
+    pub poap_instantiate_msg: POAPInstantiateMsg,
+}
+
+impl InstantiateMsg {
+    pub fn validate(&self) -> Result<(), ContractError> {
+        if self.poap_code_id == Uint64::zero() {
+            return Err(ContractError::InstantiatePOAPError {});
+        }
+        Ok(())
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    /// Allows users to claim a POAP token.
+    Claim {},
+    /// Allows the contract's admin to mint a POAP for a specific recipient.
+    MintTo { recipient: String },
+    /// Allows the contract's admin to transfer the admin rights to another user.
+    UpdateAdmin { new_admin: String },
+}
+
+impl ExecuteMsg {
+    pub fn validate(&self) -> Result<(), ContractError> {
+        Ok(())
```

Comment:
Implementation is missing here

Reply:
It is because the parameters of `ExecuteMsg` are all address string which should be verified by `deps.api.addr_validate(address)?`. I'll remove it since it does nothing.